provider/cmd/pulumi-resource-nuage/nuage_provider/ecr_repository.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2016-2021, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from typing import Any, Dict, Optional
+from dataclasses import dataclass
+
+import pulumi
+import pulumi_aws as aws
+import pulumi_random as random
+
+
+@dataclass
+class EcrRepositoryArgs:
```

Comment:
I think we should just name it Repository

If, later on, we add other cloud providers then we can use a package to namespace such as : pulumi_nuage.aws.Repository

What do you think ?